### PR TITLE
feat: add debug tool call result

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add debug tool call result

## Type of Change

- [x] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

**Review by jameszyao — APPROVED** [review]
LGTM

File: inference/providers/debug/chat_completion.py [hunk 1]
```diff
@@ -51,9 +56,12 @@ async def chat_completion(
             [function.model_dump() for function in functions] if functions else None,
             function_call,
         )
-        if provider_model_id == "debug-tool-call-hallucinations":
+        if provider_model_id == "debug-tool-call-hallucinations" and messages[-1].role == ChatCompletionRole.user:
             finish_reason = ChatCompletionFinishReason.function_calls
             message = copy.deepcopy(TOOL_CALL_HALLUCINATION_MESSAGE)
+        elif provider_model_id == "debug-tool-call-hallucinations" and messages[-1].role == ChatCompletionRole.function:
+            finish_reason = ChatCompletionFinishReason.stop
+            message = copy.deepcopy(TOOL_CALL_HALLUCINATION_RESULT)
```
Comment (jameszyao) [inline inference/providers/debug/chat_completion.py:64]: TOOL_CALL_HALLUCINATION_RESULT -> ASSISTANT_CONTENT_DEBUG_MESSAGE

**Review by jameszyao — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**